java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -56,9 +56,37 @@ const val MAX_CONSECUTIVE_FAILURES = 5
  * @property initialParticles The initial set of [Particle]s that this host contains.
  */
 abstract class AbstractArcHost(
-    private val schedulerProvider: SchedulerProvider,
+    // nullable configuration is temporary, during API transition
+    private val configuration: Configuration? = null,
     vararg initialParticles: ParticleRegistration
 ) : ArcHost {
+
+    @Deprecated("This will be removed when client migrate to primary constructor")
+    private var schedulerProvider: SchedulerProvider? = null
+
+    @Deprecated("Use primary constructor with Configuration")
+    constructor(
+        schedulerProvider: SchedulerProvider,
+        vararg initialParticles: ParticleRegistration
+    ) : this(null, *initialParticles) {
+        this.schedulerProvider = schedulerProvider
+    }
+
+    /**
+     * This is a container for platform-specific configuration objects for an [AbstractArcHost].
+     *
+     * Right now, it only contains a [HandleManagerProvider], but will eventually grow to
+     * include other configuration items.
+     */
+    data class Configuration(
+        val handleManagerProvider: HandleManagerProvider
+    ) {
+        /** Instances of this known how to provide an [EntityHandleManager]. */
+        interface HandleManagerProvider {
+            operator fun invoke(arcId: String, hostId: String): EntityHandleManager
```

Comment:
Optional nit: I'm not sure if the readability policy on operator funs, technicially this is public, does it need a KDoc? I would say no, as the interface itself documents it, but might be worth looking up or asking cameron or jwf later, i don't think it's needed.